includes/Field/BlockSupports/Anchor.php

```diff
@@ -40,11 +40,40 @@ public static function register(): void {
 		);
 	}
 
+	/**
+	 * Checks if the block spec supports the anchor field and returns the relevant interface
+	 *
+	 * @param array          $existing An existing list of a block interfaces.
+	 * @param \WP_Block_Type $block_spec The block type to register the anchor field against.
+	 *
+	 * @return string[]
+	 */
+	public static function get_block_interfaces( $existing, \WP_Block_Type $block_spec ): array {
+		if ( isset( $block_spec ) && isset( $block_spec->supports['anchor'] ) && true === $block_spec->supports['anchor'] ) {
+			$existing[] = 'BlockWithSupportsAnchor';
+		}
+		return $existing;
+	}
+
+	/**
+	 * Checks if the block spec supports the anchor field and returns the relevant interface
+	 *
+	 * @param array          $existing An existing list of a block attribute interfaces.
+	 * @param \WP_Block_Type $block_spec The block type to register the anchor field against.
+	 *
+	 * @return string[]
+	 */
+	public static function get_block_attributes_interfaces( $existing, \WP_Block_Type $block_spec ): array {
+		return self::get_block_interfaces( $existing, $block_spec );
+	}
+
 	/**
 	 * Registers an Anchor field on a block if it supports it.
 	 *
 	 * @param \WP_Block_Type $block_spec The block type to register the anchor field against.
 	 * @return void
+	 *
+	 * @deprecated 1.1.4 No longer used by internal code and not recommended.
```

Comment:
Nice addition. We can take it one step further and use `_deprecated_function()` inside the function, which will alert developers with PHP notices when WP_DEBUG is enabled. WP core uses this pattern a lot, as do many plugins. And for anyone that might be calling this method directly, it'll be a good heads up... and it gives us the chance to communicate what the 'replacement' should be, or to communicate that "we handle it automatically now" or whatever.